lib/Ast_to_Mir.ml

```diff
@@ -44,12 +44,7 @@ let targetpe e =
   let t = Var "target" in
```

Comment:
I see you throw exceptions in a couple of places, for code branches that should never happen. So far, I had been using the `fatal_error` function from `Errors.ml`. Should we have a consistent policy for this sort of thing? E.g. make `fatal_error` such that it produces a suitable error message for users to report a bug?

Reply:
Hmm, I like the Core_kernel tools, but it might be useful to have a standardized message with a link to the bugtracker. `fatal_error` is not a good name for those - there are a lot of fatal errors that are totally normal and users should NOT submit bugs, like specifying an invalid filename. Maybe we can write something like `raise_impossible` that takes in a `[%message ]` thing (looks like that's just a `Sexp.t`)?

Reply:
Sure, `raise_impossible` seems like a fine name to me. I see what you mean about not all fatal errors being the programs fault. We should just change the name of `fatal_error` I think, as I've been using it exclusively for this sort of situation. Adding a link to a bug tracker seems like a good idea.

Reply:
We also need to change the type signature to take S-Expressions, or add a fn with a `_s` suffix that does so we can use it with `[%message "bad statement" (s: stmt_loc)]` or whatever.

Reply:
Could you explain the use case of s-expressions in these errors? Is the idea that these error messages should face the developers rather than users so it makes sense to serialize internal data structures in them? Is this use of `[%message ...]` documented anywhere? I've been Googling but cannot find it. I really wish Core_kernel was documented a bit better.

Reply:
That's exactly it! I can try to find where I found this raise_s pattern soon... But yes, these are all unexpected situations the programmer failed to accommodate, so it's likely we want to serialize the offending data (as we mostly encode "situations" with data types).